test/features/showandhide/basicFields.feature

```diff
@@ -0,0 +1,39 @@
+@functional @showandhide
+Feature: Hide and show of basic fields
+
+  Background:
+    Given I have logged in
+
+  Scenario Outline: fields on the same page visible when show & hide condition is met based on mandatory and optional fields
+    Given a case type containing conditionals exists
+    And I start createCase event
+    When I enter '<value>' into the text field with id '<fieldId>'
+    Then the field with label '<expectedLabel>' is visible
+
+    Examples:
```

Comment:
was wondering whether we could add a 4th column where we display the show condition. It would help better understanding the test. Even thought is a bit of maintainability problem because we have to remember to update
that column if we change the show condition in the spreadsheet. 
Worth thinking about it

Reply:
That is why it is so time consuming to analyze the tests. We could have a comment in the .feature file saying what is in the spreadsheet for a given test. It would be better than adding an additional step.
I have had a chat with few people already. I think the solution is to generate Excel definition from the builder in the code. But this is a bigger task and we would need our PRs to host the back-end on AKS.

Reply:
This is not trivial to implement. You can easily say if the field is hidden or not by id, but the information about showCondition would have to be accessed from the Angular model.

Reply:
it's not trivial but it would be worth the investment of time in my opinion. You'll have a highly reusable step that can be used instead of many other wordy steps. That helps document the test. And it's always up to date, something a comment can't guarantee. 
I don't see why it's difficult in the step to verify 
`| visibility | show condition | initially hidden | `
Can't you grab the case type definition? easy easy

Reply:
I am not sure we are on the same page. The definition is on AAT, and the test is run in AKS and only has access to what is exposed on the UI. I cannot see how we could access `show condition`. Visibility we check by getting all labels and testing if the given label is in the array. But the show condition is in the Angular model and we execute JS agains it. If you have any ideas please share.

Reply:
not sure why this is so difficult. When we launch the suite of func tests we retrieve the case type definition once from Def Store via it's rest API `getCaseType`

Reply:
Here's what I'm suggesting:
```
Scenario Outline: fields on the same page visible when show & hide condition is met based on mandatory and optional fields
    Given a case type containing conditionals exists
    And I start createCase event
    And the page contains the following fields
    Examples:
      | fieldId               | value             | display context    | show condition                   |  shown  |
      | TextField             |                   | Mandatory          |                                  |  yes    |
      | TextFieldOptional     |                   | Optional           |                                  |  yes    |
      | TextFieldShowHide13   |                   | Optional           | TextFieldOptional="showme"       |  no     |

    When I enter '<value>' into the text field with id '<fieldId>'

    Examples:
      | value            | fieldId           |
      | showmethemoney   | TextField         |
      | showme           | TextFieldOptional |

    Then the page contains the following fields
    Examples:
      | fieldId               | value             |  shown  |
      | TextField             | showmethemoney    |  yes    |
      | TextFieldOptional     | showme            |  yes    |
      | TextFieldShowHide13   |                   |  yes    |
```